http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -0,0 +1,198 @@
+package org.broadinstitute.dsde.workbench
+package leonardo
+package util
+
+import cats.effect.Async
+import cats.mtl.Ask
+import cats.syntax.all._
+import org.broadinstitute.dsde.workbench.azure._
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.NamespaceName
+import org.broadinstitute.dsde.workbench.leonardo.SamResourceId.AppSamResourceId
+import org.broadinstitute.dsde.workbench.leonardo.config.{CoaAppConfig, SamConfig}
+import org.broadinstitute.dsde.workbench.leonardo.db.{KubernetesServiceDbQueries, _}
+import org.broadinstitute.dsde.workbench.leonardo.http._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.AppNotFoundException
+import org.broadinstitute.dsp.{Release, _}
+import org.typelevel.log4cats.StructuredLogger
+
+import java.util.Base64
+import scala.concurrent.ExecutionContext
+
+class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
```

Comment:
Most of the actual logic is in this class. There are some TODOs, but each should have a JIRA ticket.